### PR TITLE
homework week 3 - version 0.0.1

### PR DESCRIPTION
**Pull request by jawkha**
In trying to use an external database(MongoDB) for storing the contact list, I have deviated a lot from the original requirements of the work. In this iteration, we can use the API to perform basic CRUD operations. Input validation only takes place at the db level and is not comprehensive enough. In the next iteration, I will address this.
I will also update the PUT and DELETE operations to retrieve records from the database using fields other than ID.

### CODE REVIEW

File: week3/homework/src/routes/api.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+router.post('/contacts', function(req, res, next){
+    // using the create method of Mongoose (which is a promise), a new instance of the contact is created. The then method uses a callback function which carries the newly created contact as an argument, which we then use to send as a response to the client.
+    Contact.create(req.body).then(function(contact){
+        res.send(contact);
+    }).catch(next);
+});
+
+// modifies an existing contact in the contact list
+router.put('/contacts/:id', function(req, res, next){
+    Contact.findByIdAndUpdate({_id: req.params.id}, req.body).then(function(){
```
Comment (r0hitsharma) [inline week3/homework/src/routes/api.js:26]: Would you please refactor the nested `.next()` functions using Promise?

**Review by r0hitsharma — CHANGES_REQUESTED** [review]
I would really recommend you doing this homework with file instead, because I think Mongoose, being a well written library takes care of most of the conceptual issues you would've otherwise needed to consider.

For added challenge, could you try and implement an immutable `id` for the objects? You could consider two strategies: increments or hashing (this would make more sense when you work with DB's further).

**Comment by jawkha** [comment]
Hi Rohit, thanks for the feedback. I'll redo it without using MongoDB.

**Comment by jawkha** [comment]
I have uploaded a working but still unfinished version of the final project for Node module. I couldn't get back to making the requested changes for week 3 homework during this week. Will do it during the next week and upload them.

**Pull request closed without merging**